src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ManifestMetadataTableNode.cs

```diff
@@ -95,13 +101,28 @@ public int ModuleToIndex(EcmaModule module)
             {
                 if (_emissionCompleted)
                 {
-                    throw new Exception("mustn't add new assemblies after signatures have been materialized");
+                    throw new InvalidOperationException("Adding a new assembly after signatures have been materialized.");
                 }
 
+                // If we're going to add a module to the manifest, it has to be part of the version bubble, otherwise
+                // the verification logic would be broken at runtime.
+                Debug.Assert(_nodeFactory.CompilationModuleGroup.VersionsWithModule(module));
+
                 assemblyRefIndex = _nextModuleId++;
                 _manifestAssemblies.Add(assemblyName);
                 _assemblyRefToModuleIdMap.Add(assemblyName.Name, assemblyRefIndex);
             }
+            else if (_nodeFactory.CompilationModuleGroup.VersionsWithModule(module))
```

Comment:
Please consider unifying both conditional blocks - the unification should be in the above block too.

Reply:
Thanks for the suggestion Tomas. Fixed